src/pages/item/item-actions/ItemActions.tsx

```diff
@@ -55,6 +134,23 @@ export default function ItemActions(props: { itemId: string | number }) {
 				)}
 				{data && ` (${data.downstreamReferences.length})`}
 			</button>
+			<button
+				className={`button button-tertiary ${
+					isLoading ? 'button-loading button-loading-primary' : ''
+				}`}
+				onClick={showDependenciesHandler}
+				disabled={showDependenciesDisabled}
+				data-test="show-dependency"
+				title="Show Dependency & Associations"
+			>
+				{!isLoading && (
```

Comment:
Using only `isLoading` here isn't quite accurate, I think. Because in the case of the connectors, you're making additional queries after the `relationsQuery` has finished loading, namely getting the amount of cards on the board that the Item has a relation to.  
The button should be _loading_ until that operation too has ended.

Reply:
added another state called relationsLoading for this

Reply:
Mkay, but now it would make sense to aggregate these two elements here to one, because they depend on the same condition.

```jsx
        {!relationsLoading && (
          <>
            <span className="icon-add-row-bottom"></span>
            <span>Show Dependency & Associations</span>
          </>
        )}
        {!relationsLoading && ` (${relationsOnBoardCount})`}
```